### PR TITLE
Fix ipv6 iscsi live migration

### PR DESCRIPTION
**Pull request by smitterl**
The test defines an iscsi target as shared storage for guest image. The guest couldn't be started because the fileio didn't contain the guest image.
Proposal: Add test image to emulated-iscsi.

1. import data_dir:
 a. use data_dir to make vm image available on shared
    iscsi storage

2. get_virtual_size:
 a. get the virtual image size
 b. use expression e.g. "10G" because byte precision from
    image_info_dict["vsize"] led to disk out of space exception
    during libvirt.setup_or_cleanup_iscsi

3. redefine_vm_with_iscsi_target:
 a. extract method for readability

4. update_host_ip:
 a. extract method for readability

5. create_image_on_iscsi:
 a. new method to create the vm image on the shared iscsi storage

6. gluster_disk:
 a. move definition to top to avoid cleanup to fail if iscsi_setup
    fails

7. fixed several escape sequences as required by W0611

Test run on s390x:
```bash
# avocado run --vt-type libvirt --vt-machine-type s390-virtio --vt-connect-uri qemu:///system virsh.migrate_vm.positive_testing.live_migration.iscsi
JOB ID     : 7565ffb6e8394182f51ebf8bf00b20909ef3a5c0
JOB LOG    : /root/avocado/job-results/job-2019-11-04T15.52-7565ffb/job.log
 (1/4) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.live_migration.iscsi.ipv4.with_postcopy: PASS (164.14 s)
 (2/4) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.live_migration.iscsi.ipv4.without_postcopy: PASS (130.20 s)
 (3/4) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.live_migration.iscsi.ipv6.with_postcopy: PASS (147.95 s)
 (4/4) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.live_migration.iscsi.ipv6.without_postcopy: PASS (159.84 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 603.91 s
```

### CODE REVIEW

**Comment by smitterl** [comment]
@dzhengfy Could you please review? :)

File: libvirt/tests/src/migration/migrate_vm.py [hunk 1]
```diff
@@ -1055,6 +1056,69 @@ def read_domjobinfo(test, domjobinfo):
...
+    """
+    tmpdir = data_dir.get_tmp_dir()
+    emulated_path = os.path.join(tmpdir, emulated_image)
+    if vm.is_alive():
+        vm.destroy()
+    cmd = "qemu-img convert %s -O %s %s" % (disk_source, disk_format, emulated_path)
+    try:
+        result = process.run(cmd, shell=True, ignore_status=False)
+    except:
+        test.error("Couldn't prepare vm image on iscsi, %s", result)
```
Comment (chunfuwen) [inline libvirt/tests/src/migration/migrate_vm.py]: the return "result" is object,so suggest output its content.

File: libvirt/tests/src/migration/migrate_vm.py [hunk 2]
```diff
@@ -1055,6 +1056,69 @@ def read_domjobinfo(test, domjobinfo):
...
+
+
+def update_host_ip(client_ip, ipv6_addr_src):
+    """
+    Use ipv6_addr_src as host_ip if given
+    :param client_ip: Migration client ip (the host initiating the migration)
+    :param ipv6_addr_src: The vm source source host ipv6 ip
+    :return: selected host ip
+    """
+    if ipv6_addr_src:
```
Comment (chunfuwen) [inline libvirt/tests/src/migration/migrate_vm.py]: maybe can be wrapped as below:
return ipv6_addr_src if ipv6_addr_src else client_ip

Comment (smitterl) [reply libvirt/tests/src/migration/migrate_vm.py]: Thanks :) Fixed.

File: libvirt/tests/src/migration/migrate_vm.py (hunk 1 above)
Comment (smitterl) [reply libvirt/tests/src/migration/migrate_vm.py]: Thanks :) Fixed.

**Pull request merged**